scripts/evtx_dump_json.py

```diff
@@ -35,31 +35,48 @@ def main():
-                if (event_system_key=="EventRecordID"):
-                    json_subline={}
-                    firstline={event_system_key:event_system_value}
 
-                    # Add information to the JSON object for this specific log
-                    json_subline.update(firstline) #add the event ID to JSON subline
+                if not (event_system_key=="EventRecordID") or not (event_system_key=="Execution"):
+
+                    # For nested dictionaries, loop through each and extract key information 
+                    if isinstance(event_system_value,dict):
+                        for event_system_subkey,event_system_subvalue in event_system_value.items():
+
+                            if event_system_key=="EventID" or event_system_key=="TimeCreated":
+                                json_subline.update({event_system_key: event_system_subvalue})
+                            if event_system_key=="Security": 
+                                json_subline.update({event_system_subkey[1:]: event_system_subvalue})
+
+                    else: 
+                        json_subline.update({event_system_key: event_system_value})
 
             # Loop through each key, value pair of the EventData section of the evtx logs
-            for event_data_key, event_data_value in data_dict['Event']['EventData'].items():
-                for values in event_data_value:
+            if data_dict['Event']['EventData']!= None: 
+                for event_data_key, event_data_value in data_dict['Event']['EventData'].items():
+                    for values in event_data_value:
 
-                    # Loop through each subvalue within the EvenData section to extract necessary information
-                    for event_data_subkey,event_data_subvalue in values.items():
-                        if event_data_subkey=="@Name":
-                            data_name=event_data_subvalue
-                        else:
-                            data_value=event_data_subvalue
+                        # Loop through each subvalue within the EvenData section to extract necessary information
+                        for event_data_subkey,event_data_subvalue in values.items():
+                            if event_data_subkey=="@Name":
+                                data_name=event_data_subvalue
+                            else:
+                                data_value=event_data_subvalue
 
-                            # Add information to the JSON object for this specific log
-                            json_subline.update({data_name: data_value})
+                                # Add information to the JSON object for this specific log
+                                json_subline.update({data_name: data_value})
 
             # Print the JSON object for the specific log if not requested to output to file
             if not args.output:
+                print("Original")
```

Comment:
i think the output should be valid json, so it can be piped to jq, for example. so perhaps make these keys in a bigger dictionary.

Reply:
oops, that was part of my bug fix steps, will remove

Reply:
removed those statements in most recent commit and added functionality to pipe to jq 
![Screenshot from 2024-06-22 14-58-41](https://github.com/williballenthin/python-evtx/assets/71277673/22e714be-0a19-4d02-a706-d5b96179cf4d)